test/modules/http/compliance/http-errors.test.ts

```diff
@@ -0,0 +1,172 @@
+import { vi, it, expect, beforeAll, afterAll } from 'vitest'
+import http from 'http'
+import { DeferredPromise } from '@open-draft/deferred-promise'
+import { ClientRequestInterceptor } from '../../../../src/interceptors/ClientRequest'
+import { sleep, waitForClientRequest } from '../../../helpers'
+
+const interceptor = new ClientRequestInterceptor()
+
+interface NotFoundError extends NodeJS.ErrnoException {
+  hostname: string
+}
+
+interface ConnectionError extends NodeJS.ErrnoException {
+  address: string
+  port: number
+}
+
+beforeAll(() => {
+  interceptor.apply()
+})
+
+afterAll(() => {
+  interceptor.dispose()
+})
+
+it('suppresses ECONNREFUSED error given a mocked response', async () => {
```

Comment:
I've also removed the error suppression tests from the unit tests of `NodeClientRequest` and added them as compliance tests. 

In addition to that, I've added a test that ensures that it's possible to throw a connection error from the request listener and it won't be suppressed.